subprojects/core/src/main/java/org/gradle/api/internal/project/BuildOperationCrossProjectConfigurator.java

```diff
@@ -130,11 +93,27 @@ public void assertCrossProjectConfigurationAllowed(String methodName, Project ta
         return new Action<T>() {
             @Override
             public void execute(T t) {
-                nestedActionExecutionCount.set(nestedActionExecutionCount.get() + 1);
+                allowExection.set(false);
                 try {
                     action.execute(t);
                 } finally {
-                    nestedActionExecutionCount.set(nestedActionExecutionCount.get() - 1);
+                    allowExection.set(true);
+                }
+            }
+        };
+    }
+
+    // TODO: Promote to CrossProjectConfigurator interface if this is needed elsewhere.
+    public <T> Action<T> withCrossProjectConfigurationAllowed(final Action<? super T> action) {
```

Comment:
There should probably be some coverage of this method in `BuildOperationCrossProjectConfigurationTest`, too.

Reply:
done